src/mrpro/data/IData.py

```diff
@@ -154,3 +160,167 @@ def from_dicom_folder(cls, foldername: str | Path, suffix: str | None = 'dcm') -
+            mr_timing_parameters_sequence.append(Dataset())
+
+            for frame in range(number_of_frames):
+                dcm_frame_idata = dcm_file_idata[..., frame, :, :]
+                if dcm_frame_idata.header.shape.numel() != 1:
+                    raise ValueError('Only single image can be saved as a frame.')
+                directions = dcm_frame_idata.header.orientation[0].as_directions()
+                readout_direction = np.squeeze(torch.stack(directions[2].zyx[::-1]).numpy())
+                phase_direction = np.squeeze(torch.stack(directions[1].zyx[::-1]).numpy())
+                slice_direction = np.squeeze(torch.stack(directions[0].zyx[::-1]).numpy())
+                position = np.squeeze(torch.stack(dcm_frame_idata.header.position.zyx[::-1]).numpy())
+
+                dataset.PerFrameFunctionalGroupsSequence.append(Dataset())
+
+                image_position_patient = (
+                    position
+                    - readout_direction * dcm_frame_idata.header.resolution.x * dcm_frame_idata.data.shape[-1] / 2
+                    - phase_direction * dcm_frame_idata.header.resolution.y * dcm_frame_idata.data.shape[-2] / 2
+                )
+                if mr_acquisition_type == '3D':
+                    image_position_patient = (
+                        image_position_patient
+                        + slice_direction * dcm_frame_idata.header.resolution.z * (-number_of_frames / 2 + frame)
+                    )
+                if reference_patient_table_position:
+                    image_position_patient += np.squeeze(
+                        torch.stack(
+                            (dcm_frame_idata.header.patient_table_position - reference_patient_table_position).zyx[::-1]
+                        ).numpy()
+                    )
+
+                plane_position_sequence[0].ImagePositionPatient = m_to_mm(image_position_patient.tolist())
+                dataset.PerFrameFunctionalGroupsSequence[-1].PlanePositionSequence = deepcopy(plane_position_sequence)
+
+                # According to the dicom manual, ImageOrientationPatient describes:
+                # "The direction cosines of the first row and the first column with respect to the patient."
+                # This would suggest that the first direction should be the readout direction, the second the
+                # phase direction. Nevertheless, for all our data the two directions have to be swapped to achieve
+                # the correct orientation. Any help welcome in solving this!
+                plane_orientation_sequence[0].ImageOrientationPatient = [*phase_direction, *readout_direction]
+                dataset.PerFrameFunctionalGroupsSequence[-1].PlaneOrientationSequence = deepcopy(
+                    plane_orientation_sequence
+                )
+
+                def unique_parameter(parameter: torch.Tensor | list[float], parameter_name: str) -> float | None:
```

Comment:
This is not "unique", but singleton

...is this something we ned more often (util?)